samples/filters/basic-auth-filter/README.md

```diff
@@ -0,0 +1,49 @@
+# Sample for a Basic Authentication Filter
+
+Filters is a set of execution points in the request flow that intercept the request before it goes to the
+backend service. They are engaged while the request is processed within the enforcer. The defined set of filters
+are applied to all the APIs deployed in the Choreo-Connect. And these filters are engaged inline and if the request
+fails at a certain filter, the request will not be forwarded to the next filter and the backend.
+The inbuilt set of filters are the authentication filter and the throttling filter.
+
+In this sample, it would read properties called (`HeaderName`), (`Username`) and (`Password`) set at the config.toml under the filter in which the HeaderName parameter will be used to obtain contents in the relevant header and the Username and Password parameters will be authenticated against the retrieved header contents which contains the Base64 encoded username and password.
+
+# Building the Sample
+
+Maven and JDK11 is required to build the project. 
+```
+mvn clean install
+```
+
+If you are using the docker-compose setup, copy the jar file to the `docker-compose/resources/enforcer/dropins` directory,
+inside the distribution. If you are using kubernetes, make sure that this jar is mounted to the `/home/wso2/lib/dropins`
+directory of the enforcer container.
+
+Update the config.toml.
+
+```toml
+[[enforcer.filters]]
+# ClassName of the filter
+className = "org.example.tests.BasicAuth"
+# Position of the filter within final filter-chain
+position = 3
```

Comment:
Position has to be 1, I guess. Even 2 would be fine (Assuming we are using disable-security extension to disable validation).

Reply:
1 or 2 is better.

Reply:
Corrected